### PR TITLE
[Bug 1859080] Fix the bug of reporting error in azure explorer before sign in

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/mysql/MySQLModule.java [hunk 1]
```diff
@@ -43,6 +43,10 @@ public void renderChildren(List<MySqlServer> servers) {
         }
     }
 
+    protected boolean requireSignedIn() {
+        return true;
+    }
+
```
Comment (shenqianjin) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/mysql/MySQLModule.java:49]: move this method to its abstract class and default as implementation ?

**Review by shenqianjin — DISMISSED** [review]

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java [hunk 2]
```diff
@@ -136,6 +136,11 @@ public void setArcadiaModule(@NotNull HDInsightRootModule rootModule) {
         this.arcadiaModule = rootModule;
     }
 
+    @Override
+    protected void onNodeClick(NodeActionEvent e) {
+        super.onNodeClick(e);
+    }
+
```
Comment (wangmingliang-ms) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java:143]: remove it.

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/RefreshableNode.java [hunk 3]
```diff
@@ -75,15 +77,15 @@ protected void onNodeClick(NodeActionEvent e) {
...
         if (!loading) {
             setLoading(true);
             try {
                 removeAllChildNodes();
-                if (forceRefresh) {
-                    refreshFromAzure();
+                if (AuthMethodManager.getInstance().isSignedIn() || this instanceof AzureModule) {
+                    refreshItems();
                 }
-                refreshItems();
```
Comment (wangmingliang-ms) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/RefreshableNode.java:86]: these 2 methods is meaningful in old implementation:
`refreshFromAzure` requires reload the data from Azure(when user click the `fresh` button/menu item).
`refreshItems` indicates we can load from cache(when user click to expand a parent node.)

Comment (andxu) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/RefreshableNode.java:86]: refreshFromAzure  are not used in current impl, remove it and will add it if we need it.

**Review by wangmingliang-ms — APPROVED** [review]

**Review by shenqianjin — APPROVED** [review]
LGTM

**Pull request merged**